src/legacy/core_plugins/kibana/public/visualize/wizard/new_vis_modal.tsx

```diff
@@ -130,16 +130,18 @@ class NewVisModal extends React.Component<TypeSelectionProps, TypeSelectionState
 
   private redirectToVis(visType: VisType | VisTypeAlias, searchType?: string, searchId?: string) {
     this.trackUiMetric(METRIC_TYPE.CLICK, visType.name);
-
+    let params;
     if ('aliasUrl' in visType) {
-      window.location.assign(this.props.addBasePath(visType.aliasUrl));
+      params = this.props.addBasePath(visType.aliasUrl);
       if (this.props.editorParams && this.props.editorParams.includes('addToDashboard')) {
+        params = `${params}?addToDashboard`;
```

Comment:
This means maps will be called with `addToDashboard` param. If that's an issue, this can be an explicit check for Lens.